### DISCUSSION TITLE
NoExplicitTypeError on simple types

### DISCUSSION THREAD

**Discussion by gederer** [MichalLytek/type-graphql#1094] [Q&A]
I'm getting this on all properties decorated with `@Field()`:

`NoExplicitTypeError: Unable to infer GraphQL type from TypeScript reflection system. You need to provide explicit type for 'email' of 'AddUserInput' class.`

I would expect to get this on `Object` and `Array` properties. But, I'm getting it with `string`s and `number`s, too.

E.g.:

```
@InputType()
class AddUserInput {
  @Field()
  email: string;

  @Field()
  username: string;

  @Field()
  password: string;

  @Field()
  name: string;
}
```

Any suggestions? Thanks.

**Comment by MichalLytek** [comment]
https://typegraphql.com/docs/installation.html#typescript-configuration

**Reply by gederer** [reply]
Thanks for the reply. Thought it might have something to do with tsconfig.

Now I'm getting:

```
    __metadata("design:type", User)
                              ^

ReferenceError: Cannot access 'User' before initialization
```

The transpiled code looks like this:

```
__decorate([
    Field(() => User),
    __metadata("design:type", User)
], Post.prototype, "author", void 0);
```

**Comment by KeithAlt** [comment]
Encountered this error and fixed by verifying my tsconfig.
Specifically was missing `"emitDecoratorMetadata": true`.

**Comment by 0xborisgregor** [comment]
For anyone who already configured 
```json
{
  "emitDecoratorMetadata": true,
  "experimentalDecorators": true
}
```
 in `tsconfig.json` but still run into `NoExplicitTypeError`, you might be using `tsx` which doesn't support `emitDecoratorMetadata`: [tsx compiler limitations](https://tsx.is/typescript#compiler-limitations)
